src/deepsparse/server/config.py

```diff
@@ -12,152 +12,73 @@
-from typing import List
+from typing import List, Optional, Tuple, Union
 
-import yaml
 from pydantic import BaseModel, Field
 
-from deepsparse import PipelineConfig
-from deepsparse.cpu import cpu_architecture
-
 
 __all__ = [
-    "ENV_DEEPSPARSE_SERVER_CONFIG",
-    "ENV_SINGLE_PREFIX",
     "ServerConfig",
+    "EndpointConfig",
+    "SequenceLengthsConfig",
+    "ImageSizesConfig",
 ]
 
 
-ENV_DEEPSPARSE_SERVER_CONFIG = "DEEPSPARSE_SERVER_CONFIG"
-ENV_SINGLE_PREFIX = "DEEPSPARSE_SINGLE_MODEL:"
+class SequenceLengthsConfig(BaseModel):
+    sequence_lengths: List[int] = Field(
+        description="The sequence lengths the model should accept"
+    )
 
 
-class ServerConfig(BaseModel):
-    """
-    A configuration for serving models in the DeepSparse inference server
-    """
-
-    models: List[PipelineConfig] = Field(
-        default=[],
-        description=(
-            "The models to serve in the server defined by PipelineConfig objects"
-        ),
+class ImageSizesConfig(BaseModel):
+    image_sizes: List[Tuple[int, int]] = Field(
+        description="The list of image sizes the model should accept"
     )
-    workers: str = Field(
-        default=max(1, cpu_architecture().num_available_physical_cores // 2),
-        description=(
-            "The number of maximum workers to use for processing pipeline requests. "
-            "Defaults to the number of physical cores on the device."
-        ),
+
+
+class EndpointConfig(BaseModel):
+    name: str = Field(
+        description="Name of the model used for logging & metric purposes."
     )
-    integration: str = Field(
-        default="default",
-        description=(
-            "Name of deployment integration that this server will be deployed to "
-            "Currently supported options are None for default inference and "
-            "'sagemaker' for inference deployment with AWS Sagemaker"
-        ),
+
+    endpoint: str = Field(
+        description="The path to use for this endpoint. E.g. '/predict'."
     )
 
+    task: str = Field(description="Task this endpoint performers")
+
+    model: str = Field(description="Location of the underlying model to use.")
+
+    batch_size: int = Field(description="The batch size to compile the model for.")
+
+    accept_multiples_of_batch_size: bool = Field(
+        description="""
+Whether to accept any request with a batch size that is a multiple of `batch_size`.
+
+E.g. if batch_size is 1 and this field is True,
+then the model can accept any batch size.
+    """,
+    )
+
+    bucketing: Optional[Union[ImageSizesConfig, SequenceLengthsConfig]] = Field(
+        default=None,
+        description="What input shapes this model can accept. Must specify at least 1",
+    )
+
+
+class ServerConfig(BaseModel):
+    num_cores: int = Field(
+        description="The number of cores available for model execution. "
+        "Defaults to all available cores.",
+    )
+
+    num_concurrent_batches: int = Field(
```

Comment:
prefer something closer to num_works/num_streams over num_concurrent_batches. additionally we'll need an engine doc on tuning this